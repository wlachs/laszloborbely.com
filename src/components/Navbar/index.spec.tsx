import { cleanup, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { afterEach, describe, expect, it, vi } from 'vitest';

import { Navbar } from './';

const mocked = vi.hoisted(() => ({
	component: () => <div />,
}));

vi.mock('../Links', () => ({
	Links: mocked.component,
}));

vi.mock('../PureLinkHighlight', () => ({
	PureLinkHighlight: mocked.component,
}));

describe('Navbar component', () => {
	afterEach(() => {
		cleanup();
	});

	it('should match snapshot', () => {
		const { container } = render(
			<MemoryRouter>
				<Navbar />
			</MemoryRouter>,
		);
		expect(container).toMatchSnapshot();
	});
});
