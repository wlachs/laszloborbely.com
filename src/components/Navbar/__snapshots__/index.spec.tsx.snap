// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar component > should match snapshot 1`] = `
<div>
  <div
    class="flex items-center justify-between py-4"
  >
    <div />
    <div
      class="flex flex-row-reverse items-center gap-3"
    >
      <div />
    </div>
  </div>
  <div
    class="separator w-100 mb-3"
  />
</div>
`;
